### PR TITLE
[C] make InternalChildren public on Layout so it can be accessed the same way as we do on Page

### PR DESCRIPTION
**Pull request by ChaseFlorell**
### Description of Change ###

This change brings alignment between `Page` and `Layout` in as much as being able to access `InternalChildren` for the purposes of accessing elements within a ControlTemplate.

Currently when I create a new ContentPage and add a `<ControlTemplate>`, I have the ability to locate controls within that ControlTemplate. 

This change will allow us to do the same thing off of a ContentView or similar View using `Layout`

```xaml
<ContentPage >
    <ContentPage.ControlTemplate>
        <ControlTemplate>
            <AbsoluteLayout >
                <StackLayout x:Name="MyStack">
                    <Label x:Name="MyLabel" />
                </StackLayout> 
            </AbsoluteLayout>
        </ControlTemplate>
    </ContentPage.ControlTemplate>
</ContentPage>
```
```csharp
private Label _myLabel;
ctor()
{
    InitializeComponent();
    var root = (AbsoluteLayout)InternalChildren.First();
    _myLabel = root.FindByName<StackLayout>("MyStack")
                   .FindByName<Label>("MyLabel");
}

protected override void OnAppearing()
{
    _myLabel.Text = "See, I can access the label from OnAppearing";
}
```

### Issues Resolved ### 

- n/a - no issue filed

### API Changes ###

Changed:
 - internal Layout.InternalChildren => public Layout.InternalChildren
 

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by ChaseFlorell** [comment]
As a side note, I'd love to see this property get renamed to something more suitable and not indicative of being "Internal". It's fantastic to be able to locate content within a ControlTemplate.

File: Xamarin.Forms.Core/Layout.cs [hunk 1]
```diff
@@ -104,8 +104,12 @@ void IPaddingElement.OnPaddingPropertyChanged(Thickness oldValue, Thickness newV
 		{
 			InvalidateLayout();
 		}
-
-		internal ObservableCollection<Element> InternalChildren { get; } = new ObservableCollection<Element>();
+		
+		/// <summary>For internal use by the Xamarin.Forms platform.</summary>
+		/// <value>To be added.</value>
+		/// <remarks>To be added.</remarks>
+		[EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Layout.cs]: your comment is in violation with your example.

**Comment by ChaseFlorell** [comment]
It's already there on `Page`.
The biggest reason I want this it to apply animations to elements within a Control Template

**Comment by ChaseFlorell** [comment]
> your comment is in violation with your example.

Assuming you're referring to the code comment, it's just there to match Page. I would hope to spark a discussion around making it more available with a new property name (leaving the existing one obsolete)

File: Xamarin.Forms.Core/Layout.cs (hunk 1 above)
Comment (ChaseFlorell) [reply Xamarin.Forms.Core/Layout.cs]: It's just there to match Page. I would hope to spark a discussion around making it more available with a new property name (leaving the existing one obsolete).

What do you suggest?

Comment (ChaseFlorell) [reply Xamarin.Forms.Core/Layout.cs]: @StephaneDelcroix bump.
What would you suggest to fix this?

Comment (ChaseFlorell) [reply Xamarin.Forms.Core/Layout.cs]: How can I improve the comment?

Comment (kingces95) [reply Xamarin.Forms.Core/Layout.cs]: Just erase the comments per `Page`.

**Comment by ChaseFlorell** [comment]
I disagree @StephaneDelcroix 
I build Custom ContentViews with a ControlTemplate all the time, and every time I need to access a control within, I have to unwind the ControlTemplate. It's super aggravating knowing that it's accessible at the Page level.

**Comment by StephaneDelcroix** [comment]
@ChaseFlorell I could understand your need to access the realized content of a ControlTemplated view, even if TemplateBinding should solve most of your issues. accessing a layout and then adding views in it... I'd argue that messing with the expanded template defeats the separation of concerns it should bring...

but opening InternalsChildren is not the right way to do it. but I'm open to discuss adding a new property, or helper, to access the realized view. Feel free to submit a proposal for that.

**Comment by ChaseFlorell** [comment]
Well honestly, I'd like `ControlTemplate` to include a `FindByName<T>()` and be done with it.

```xml
<ContentPage >
    <ContentPage.ControlTemplate>
        <ControlTemplate x:Name="MyTemplate">
            <AbsoluteLayout >
                <StackLayout x:Name="MyStack">
                    <Label x:Name="MyLabel" />
                </StackLayout> 
            </AbsoluteLayout>
        </ControlTemplate>
    </ContentPage.ControlTemplate>
</ContentPage>
```

```csharp
var label = MyTemplate.FindByName<StackLayout>("MyStack")
                      .FindByName<Label>("MyLabel");
label.Text = "See, I can access the label from OnAppearing";
```

**Comment by samhouts** [comment]
@ChaseFlorell Thanks for the suggestion!! Let's discuss this on the issue I just opened for you. Thanks!!

**Pull request closed without merging**